src/components/svgs/lines/styles/mrt-under-construction.tsx

```diff
@@ -0,0 +1,79 @@
+import { RmgFields, RmgFieldsField } from '@railmapgen/rmg-components';
+import { CityCode, MonoColour } from '@railmapgen/rmg-palette-resources';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { AttrsProps } from '../../../../constants/constants';
+import {
+    LinePathAttributes,
+    LinePathType,
+    LineStyle,
+    LineStyleComponentProps,
+    LineStyleType,
+} from '../../../../constants/lines';
+import { AttributesWithColor, ColorField } from '../../../panels/details/color-field';
+
+const MRTUnderConstruction = (props: LineStyleComponentProps<MRTUnderConstructionAttributes>) => {
+    const { id, path, styleAttrs, handleClick } = props;
+    const { color = defaultMRTUnderConstructionAttributes.color } = styleAttrs ?? defaultMRTUnderConstructionAttributes;
+
+    const onClick = React.useCallback(
+        (e: React.MouseEvent<SVGPathElement, MouseEvent>) => handleClick(id, e),
+        [id, handleClick]
+    );
+
+    return (
+        <g id={id}>
+            <path d={path} fill="none" stroke={color[2]} strokeWidth="5" strokeDasharray="0 10" strokeLinecap="round" />
+            <path
+                d={path}
+                fill="none"
+                stroke="white"
+                strokeOpacity="0"
+                strokeWidth={5}
+                cursor="pointer"
+                onClick={onClick}
+            />
+        </g>
+    );
+};
+
+/**
+ * MRTUnderConstruction specific props.
+ */
+export interface MRTUnderConstructionAttributes extends LinePathAttributes, AttributesWithColor {}
+
+const defaultMRTUnderConstructionAttributes: MRTUnderConstructionAttributes = {
+    color: [CityCode.Beijing, 'bj1', '#c23a30', MonoColour.white],
```

Comment:
Maybe Singapore/ewl.

Reply:
Speaking of automatically overwriting the runtime color to the elements' color, I do have issue #478 for it. But the question is how do we provide a switch for users as both approaches are needed in different cases.

Reply:
A check box beside 'Color' in the side bar?

If we do this, we can add a select box to choose line style type.

Reply:
Or some tweaks in settings. Not sure what the percentage of users who would like to overwrite or use the default color.

Having much on the tools panel may make it hard to use. Users may lost on so many options and we could leave some advanced/professional features in places not easy to find.

Reply:
Let's move more replies to #478 and merge this now :)